vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -308,116 +317,148 @@ def get_external_host_tags(self):
 
         return external_host_tags
 
-    def _discover_mor(self, instance, tags, regexes=None, include_only_marked=False):
+    def _get_parent_tags(self, mor, all_objects):
+        tags = []
+        properties = all_objects[mor]
+        parent = properties["parent"]
+        if parent:
+            parent_name = all_objects[parent]["name"]
+            tag = []
+            if isinstance(parent, vim.HostSystem):
+                tag.append(u'vsphere_host:{}'.format(parent_name))
+            elif isinstance(parent, vim.Folder):
+                tag.append(u'vsphere_folder:{}'.format(parent_name))
+            elif isinstance(parent, vim.ComputeResource):
+                if isinstance(parent, vim.ClusterComputeResource):
+                    tag.append(u'vsphere_cluster:{}'.format(parent_name))
+                tag.append(u'vsphere_compute:{}'.format(parent_name))
+            elif isinstance(parent, vim.Datacenter):
+                tag.append(u'vsphere_datacenter:{}'.format(parent_name))
+
+            tags = self._get_parent_tags(parent, all_objects)
+            if tag:
+                tags.extend(tag)
+
+        return tags
+
+    def _collect_mors_and_attributes(self, server_instance, max_objects):
```

Comment:
If you move this method before `_get_parent_tags ` then the git diff may be better to understand your changes.

Reply:
I'm not sure it would. I removed the nested functions from the `_discover_mors` method, and remove it. So the diff will be a mess anyway I think.